internal/services/cdn/cdn_frontdoor_custom_domain_association_resource_test.go

```diff
@@ -0,0 +1,242 @@
+package cdn_test
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform-provider-azurerm/internal/acceptance"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/acceptance/check"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/clients"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/services/cdn/parse"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/utils"
+)
+
+type CdnFrontDoorCustomDomainAssociationResource struct {
+}
+
+// NOTE: There isn't a complete test case because the basic and the
+// update together equals what the complete test case would be...
+func TestAccCdnFrontDoorCustomDomainAssociation_basic(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_cdn_frontdoor_custom_domain_association", "test")
+	r := CdnFrontDoorCustomDomainAssociationResource{}
+
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.basic(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+	})
+}
+
+// NOTE: the 'requiresImport' test is not possible on this resource
+
+func TestAccCdnFrontDoorCustomDomainAssociation_removeAssociation(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_cdn_frontdoor_custom_domain_association", "test")
+	r := CdnFrontDoorCustomDomainAssociationResource{}
+
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.basic(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		{
+			Config:             r.remove(data),
+			Check:              acceptance.ComposeTestCheckFunc(),
+			ExpectNonEmptyPlan: true, // since deleting this resource actually removes the linked custom domain from the route resource(s)
+		},
+	})
+}
+
+func TestAccCdnFrontDoorCustomDomainAssociation_removeAssociations(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_cdn_frontdoor_custom_domain_association", "test")
+	r := CdnFrontDoorCustomDomainAssociationResource{}
+
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.update(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		{
+			Config:             r.remove(data),
+			Check:              acceptance.ComposeTestCheckFunc(),
```

Comment:
I can't because the state in the test isn't updated and the value comes back incorrect that is with the non-empty plan check is for. If it has a non empty plan that means that the resource did change the value on the associated route resource.